constantdict/__init__.py

```diff
@@ -170,3 +188,96 @@ def discard(self, key: K) -> constantdict[K, V]:
     setdefault = _del_attr  # type: ignore[assignment]
 
     # }}}
+
+    # {{{ mutation
+
+    def mutate(self) -> constantdictmutation[K, V]:
+        """Return a mutable copy of this :class:`constantdict` as a
+        :class:`constantdictmutation`.
+
+        Run :meth:`constantdictmutation.finish` to convert back to an immutable
+        :class:`constantdict`.
+
+        .. note::
+
+            Based on the `immutables.Map API <https://github.com/MagicStack/immutables>`__.
+
+        .. doctest::
+
+            >>> cd = constantdict(a=1, b=2)
+            >>> cd_mut = cd.mutate()
+            >>> cd_mut["a"] = 10
+            >>> cd_new = cd_mut.finish()
+            >>> cd_new
+            constantdict({'a': 10, 'b': 2})
+            >>> cd
+            constantdict({'a': 1, 'b': 2})
+        """
+        # This needs to make a copy since the original dictionary must
+        # not be modified.
+        return constantdictmutation(self)
+
+    # }}}
+
+
+class constantdictmutation(constantdict[K, V]):  # noqa: N801
```

Comment:
Perhaps it's better to derive this directly from `dict`, instead of `constantdict`, and just provide the extra `finish()` method.

Reply:
Done in [02fcd2b](https://github.com/matthiasdiener/constantdict/pull/20/commits/02fcd2b398dcb1fa4fe21df85950f56e9c5d584b)